conjure-java-core/src/main/java/com/palantir/conjure/java/types/AliasGenerator.java

```diff
@@ -215,16 +222,32 @@ private static CodeBlock valueOfFactoryMethodForPrimitive(
                 "Unsupported primitive type: " + primitiveType + "for `valueOf` method.");
     }
 
+    /**
+     * Detects if the class is available in the generator classpath and has a public static <code>valueOf(String)
+     * </code>.
+     */
+    private static boolean hasValueOfFactory(com.palantir.conjure.spec.TypeName className) {
```

Comment:
I believe that with this change we would be implicitly exposing the dependencies of conjure-java as API since changes to the dependencies could impact the generated code. What case did you have in mind for this?

Reply:
I have several `alias: Long` where `Long` is an external-type-import of java.lang.Long. I can bound this to primitives.